### DISCUSSION TITLE
Display distance from user location

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#338] [Ideas]
During the conversation in the neighboring topic, I wanted to ask for the ability to show the (approximate) distance between the center of the map and the current location. 

In other applications it is not always readable. In Cruiser we have a nice pulsating field on the edge, towards us. Great stuff! Maybe supplement them with this distance?

**Comment by devemux86** [comment]
@Ciutnik thanks for the suggestion.

Can you post some screenshots?

**Reply by Ciutnik** [reply]
This is what it looks like in Osmand and Locus. In our case, the line is not needed, since we have a pulsating field, and the value in the panel... is probably not needed.

<img src="https://github.com/devemux86/cruiser/assets/122647782/4b659358-141a-4b89-8e25-e848450fd95f" width="49%"/>

<img src="https://github.com/devemux86/cruiser/assets/122647782/2fc85d8f-6f03-46be-8c6b-856fad9e26f2" width="49%"/>

**Reply by devemux86** [reply]
@Ciutnik thanks for the screenshots.

The panel seems more readable, however what is its use if it is only used when temporarily scrolling the map?

**Reply by Ciutnik** [reply]
The problem with the panel is that it takes many clicks to turn it on/off. And yet this type of information is only important for a moment, so maybe it could only be visible when swiping map (and a moment after)? Or when we are just in an offset position? Maybe a temporary bottom panel?

**Reply by kds112** [reply]
Maybe this distance panel can have the selections Always/Never/Larger than 0/Larger than X meters.
Maybe we need the last selection because of GPS inaccuracy and I don't want to have a panel that is switching on and off permanently...

**Reply by kds112** [reply]
Or the selections are always / never / current position out of displayed map...

**Reply by Ciutnik** [reply]
I'm not sure if the "Always" option is needed. Most of the time, when navigating or following, it would show "0". I think it's only necessary when you move the map manually.

**Comment by devemux86** [comment]
We can also display the _altitude delta_ in addition to the distance from the user location.
This would be useful in many profiles.

**Reply by kds112** [reply]
Is it possible to re-introduce this blue dot with mapsforge rendering as well? Unfortunately, it works with OpenGL only.

**Reply by devemux86** [reply]
@kds112 This is an OpenGL effect.
Not available in Mapsforge, unless we implement something new.

**Reply by kds112** [reply]
Yes, I know. And might it be possible to implement something similar in Mapsforge?  Should I open a new topic for this?

**Reply by devemux86** [reply]
@kds112 Please create a new thread so we don't forget about it here.

The feature being discussed here, we don't yet know its implementation, panels or something else.
